galpy/df/streamspraydf.py

```diff
@@ -297,6 +295,56 @@ def _setup_rot(self, dt):
         rot_inv = numpy.einsum("ijk,ikl->ijl", z_rot_inv, pa_rot_inv)
         return (rot, rot_inv)
 
+    def _calc_rtide_vcs(self, Rpt, phipt, Zpt, dt):
+        try:
+            rtides = rtide(
+                self._rtpot,
+                Rpt,
+                Zpt,
+                phi=phipt,
+                t=-dt,
+                M=self._progenitor_mass,
+                use_physical=False,
+            )
+            vcs = numpy.sqrt(
+                -Rpt
+                * evaluateRforces(
+                    self._rtpot, Rpt, Zpt, phi=phipt, t=-dt, use_physical=False
+                )
+            )
+        except (ValueError, TypeError):
+            rtides = numpy.array(
+                [
+                    rtide(
+                        self._rtpot,
+                        Rpt[ii],
+                        Zpt[ii],
+                        phi=phipt[ii],
+                        t=-dt[ii],
+                        M=self._progenitor_mass,
+                        use_physical=False,
+                    )
+                    for ii in range(len(Rpt))
+                ]
+            )
+            vcs = numpy.array(
+                [
+                    numpy.sqrt(
+                        -Rpt[ii]
+                        * evaluateRforces(
+                            self._rtpot,
+                            Rpt[ii],
+                            Zpt[ii],
+                            phi=phipt[ii],
+                            t=-dt[ii],
+                            use_physical=False,
+                        )
+                    )
+                    for ii in range(len(Rpt))
+                ]
+            )
+        return rtides, vcs
```

Comment:
Since in your method you only need the tidal radius, not the circular velocity, it would be somewhat more efficient to have two methods, one for the tidal radius and one for the circular velocity, so you can just use the tidal radius one.